presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionFunction.java

```diff
@@ -158,4 +170,11 @@ public static long arrayPosition(
         }
         return 0;
     }
+
+    private static void checkNotIndeterminate(Boolean equalsResult)
+    {
+        if (equalsResult == null) {
+            throw new PrestoException(NOT_SUPPORTED, "array_position does not support arrays with elements that are null or contain null");
```

Comment:
Why `array_position` would not support arrays with nulls? It makes sense to not support `array_position(a, null)`, but it's ok for the array to have null right? Actually I think that is the case cause there is `if (!array.isNull(i))`. So as long as we check `element` is not null we don't need to do this check here?

Reply:
There was a controversy of what the arrayPostition has to return in case when EQUALS returns that the result is indeterminate.

> array_position(a, null)

In this case the `arrayPosition` method returns null automatically, as the element is not marked as nullable and doesn't accept nulls: https://github.com/prestodb/presto/pull/10909/files/37aaf4978533dfe625a7084af815c560bf9b09eb..e16cb85cbf116e88ced4722a18efe8ace6f7b4d6#diff-16a3f69f8928c469717238b977cf04bbR45

If the arrays is itself `null`, the method also returns `null` automatically: https://github.com/prestodb/presto/pull/10909/files/37aaf4978533dfe625a7084af815c560bf9b09eb..e16cb85cbf116e88ced4722a18efe8ace6f7b4d6#diff-16a3f69f8928c469717238b977cf04bbR44


The only case when the method is invoked is when the both, haystack (the array) and the needle (element) are not null.

Assuming that the needle (element) is not null the only possibility of indeterminism is when some element in a haystack is null.

e.g.: `array_position(ARRAY[null, 1, 2], 1)`

However, based on this line (https://github.com/prestodb/presto/pull/10909/files/37aaf4978533dfe625a7084af815c560bf9b09eb..e16cb85cbf116e88ced4722a18efe8ace6f7b4d6#diff-16a3f69f8928c469717238b977cf04bbR49) the situation when the top level element in the array is null is explicitly checked, and such position is skipped.

So the check for indeterminism is actually about queries like 

> array_position(ARRAY[ARRAY[1, 2], ARRAY[1, NULL]], ARRAY[1, NULL])

In such case the EQUALS(ARRAY[1, NULL], ARRAY[1, NULL]) returns that the result is indeterminate.

The controversy was how to interpret such a situation? Should it be interpreted as a match? not match? null?

Reply:
In that case the check is only useful when the element type is `Block`. We can remove it from all other implementations?

Reply:
And a better error message might be "array_position does not support element of array that contains null"?